### PR TITLE
Make Animated Value persist its value after reattaching

### PR DESCRIPTION
**Pull request by osdnk**
When value gets detached from all component it can be still kept in memory and attached to some other component at later time. But because we don't know when JS reference is dropped we need to delete native counterpart of the value when it gets detached. This has led to a situation in which the value was reinstantiated on the native site. In that case if the value has changed we expect it to persist that changes when it gets attached again. This wasn't happening and each time the value was instantiated we'd always use the initial value.

With this change we are solving that issue by querying the value right before we detach the native node and then we use that value to update config such that next time when it gets instantiated it uses updated value instead of the initial one.

### CODE REVIEW

**Comment by osdnk** [comment]
related: https://github.com/kmagiera/react-native-reanimated/issues/85

**Comment by osdnk** [comment]
@kmagiera 
Now I am not convinced at all about this change. I'm afraid of blinking with some edge cases like this one which Eric has shown

**Comment by ckknight** [comment]
@kmagiera Why use a callback instead of returning a promise?

File: src/core/AnimatedValue.js [hunk 1]
```diff
@@ -3,6 +3,8 @@ import { set } from '../base';
 import { val } from '../utils';
 import { evaluateOnce } from '../derived/evaluateOnce';
 import interpolate from '../derived/interpolate';
+import ReanimatedModule from '../ReanimatedModule';
+import ReanimatedEventEmitter from '../ReanimatedEventEmitter';
```
Comment (kmagiera) [inline src/core/AnimatedValue.js]: import cleanup

**Pull request merged**